submissions/oluwatosinOke/slimPhp/app/controllers/ArticlesController.php

```diff
@@ -0,0 +1,131 @@
+use Psr\Http\Message\ResponseInterface as Response;
+use Psr\Http\Message\ServerRequestInterface as Request;
+
+session_start();
+
+class ArticleController
+{
+    public function index(Request $request, Response $response)
+    {
+        $articles = Article::all();
+
+        $response->getBody()->write(json_encode(["status" => "success", "data" => [$articles]]));
+        return $response->withHeader('Content-Type', 'application/json')
+            ->withStatus(200);
+    }
+
+    public function create(Request $request, Response $response)
+    {
+        if (!isset($_SESSION['admin'])) {
+            $response->getBody()->write(json_encode(['error' => 'Only an authenticated admin can perform this action']));
+            return $response->withHeader('Content-Type', 'application/json')->withStatus(403);
+        };
+        $data = $request->getParsedBody();
+        $createdAt = date('Y-m-d H:i:s');
+        $id = $_SESSION["admin"];
+        $article = Article::create([
+            'title' => $data['title'],
+            'description' => $data['description'],
+            "status" => "unpublished",
+            "created_by" => $id,
+            "created_at" => $createdAt,
+        ]);
+
+        $article->save();
+
+        $response->getBody()->write((json_encode(["status" => "success", "data" => $article])));
+        return $response->withHeader('Content-Type', 'application/json')
+            ->withStatus(200);
+    }
+
+    public function update(Request $request, Response $response, $args)
+    {
+        $data = $request->getParsedBody();
+        $id = $args['id'];
+
+        $article = Article::find($id);
+        if (!$article) {
+            $response->getBody()->write(json_encode(['error' => 'Article not found']));
+            return $response->withHeader('Content-Type', 'application/json')->withStatus(404);
+        }
+
+        $article->title = $data['title'];
+        $article->description = $data['description'];
+
+        $article->save();
+
+        $response->getBody()->write(json_encode(["status" => "success", "data" => $article]));
+        return $response->withHeader('Content-Type', 'application/json')
+            ->withStatus(200);
+    }
+
+    public function delete(Request $request, Response $response, $args)
+    {
+        $data = $request->getParsedBody();
+        $id = $args['id'];
+
+        $article = Article::find($id);
+        if (!$article) {
+            $response->getBody()->write(json_encode(['error' => 'Article not found']));
+            return $response->withHeader('Content-Type', 'application/json')->withStatus(404);
+        }
+
+        $article->delete();
+
+        $response->getBody()->write(json_encode(["status" => "success", "message" => "This article was succssfully deleted"]));
+        return $response->withHeader('Content-Type', 'application/json')
+            ->withStatus(200);
+    }
+
+    public function publish(Request $request, Response $response, $args)
```

Comment:
I am very impressed with the way you handled this endpoint.

Reply:
I can't find the link to your documentation. KIndly place the link on your Readme file